### DISCUSSION TITLE
How to get ambiguity fractional cycles of fixed ones in scenarioLeoReducedDynamicOrbit and scenarioLeoKinematicOrbit process?

### DISCUSSION THREAD

**Discussion by lorangedd** [groops-devs/groops#207] [Q&A]
Dear groops team,

In an example of GRACE-FO by scenarioLeoReducedDynamicOrbit XML,  I did the writeAprioriSolution before resolveAmbiguities and got the float ambiguity of L1 and L2, their distributions are as following picture.

![image](https://github.com/groops-devs/groops/assets/45618570/b15add00-39d9-4a5b-acf7-feb8c73b594c)
![image](https://github.com/groops-devs/groops/assets/45618570/79b6fc24-5002-41ca-81e8-b4e5a6587d8c)

Obviously, the float L1 ambiguity fractional cycles are not normal distribution as L2, even so, the final orbit is still accurate, Why?

As we know, the integer ambiguity is pre-eliminated after resolveAmbiguities , so how can we acquire the fractional part of the fixed ambiguities by modifying the source code?

I would be very happy to receive your feedback on these ideas.

**Comment by tmayerguerr** [comment]
Dear lorangedd,

The estimated float ambiguities are not the L1/L2 but transformed with the LAMBDA method for numerical stability. Therefore the parameters are something like the wide-lane and narrow-lane ambiguities. In the ambiguity fixing process, the more precise ambiguities are fixed first. This makes the other ambiguities more precise and they should then also be normally distributed.

Regarding your question: What do you mean with "fractional part of the fixed ambiguities"?
After the ambiguity resolution the integer ambiguities are directly reduced from the observations and not setup as parameters
in the following processing steps. The remaining fractional part is called phase bias in GROOPS and the outputfile can be defined in [parametrization:SignalBias](https://groops-devs.github.io/groops/html/gnssParametrizationType.html#signalBiases). Use [processingStep:writeResults](https://groops-devs.github.io/groops/html/gnssProcessingStepType.html#writeResults). I hope this answers your question.

regards
Torsten

**Reply by lorangedd** [reply]
Thank you for your response. 

First, for "fractional part of the fixed ambiguities", it means the difference between fixed integer ambiguity cycles and float ambiguities before the ambiguity resolution.

Then, about the estimated float ambiguities, I can't understand "something like the wide-lane and narrow-lane ambiguities", is there any reference or further explanation about it?


regards
Li

**Reply by Hanbing59** [reply]
May be a late reply, and you have already found your answers, but for the references of others:

"wide-lane and narrow-lane ambiguities" in @tmayerguerr's reply, they refer to concepts in single-station ambiguity fixing for dual-frequency ionosphere-free phase observations, for which, a classic and highly cited paper is [Ge et al. 2008](https://link.springer.com/article/10.1007/s00190-007-0187-4). You may start from there.
